cmd/gcp-controller-manager/app/loops.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"sort"
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+	"golang.org/x/oauth2"
+	compute "google.golang.org/api/compute/v1"
+	gcfg "gopkg.in/gcfg.v1"
+	warnings "gopkg.in/warnings.v0"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce"
+	"k8s.io/kubernetes/pkg/controller/certificates"
+)
+
+type ControllerContext struct {
+	client                      clientset.Interface
+	sharedInformers             informers.SharedInformerFactory
+	recorder                    record.EventRecorder
+	gcpCfg                      GCPConfig
+	clusterSigningGKEKubeconfig string
+	stopCh                      <-chan struct{}
+}
+
+func (sc *ControllerContext) Client() clientset.Interface {
```

Comment:
Make all fields exported instead of all the methods

Reply:
I removed all of these methods. They don't even need to be exported.